### DISCUSSION TITLE
Enhancing Github Copilot's Context Awareness

### DISCUSSION THREAD

**Discussion by VictorMaderaA** [community/community#50338] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Hello everyone!

As a developer who frequently uses Github Copilot and Jetbrains IDE, I wanted to provide some feedback to the team regarding Copilot's contextual awareness. While Copilot has been an incredibly useful tool in my coding process, there are times when it lacks context for certain lines of code.

Therefore, I would like to suggest a future development for Copilot to be able to take context from opened tabs in the IDE. This would allow Copilot to not only read the latest modified files or lines written, but also grab context from the opened tabs. For larger projects with thousands of files, this would enable Copilot to prioritize the files that matter for the current suggestion.

I am not sure if this feature is already on the roadmap, or if it has been implemented already, but I hope my feedback will be considered. Overall, I am grateful for the amazing work the Copilot team has done and I look forward to seeing how this tool will continue to evolve in the future.

**Comment by WinkelCode** [comment]
It already has that capability in VSCode. However, what it can see exactly is a bit unclear and inconsistent. In the past I've had success using a "style guide" file when working with copilot.

![image](https://user-images.githubusercontent.com/29005772/226149882-be18a550-f8de-4293-9b11-f6fc7e8e624e.png)
